### ISSUE TITLE
On opt-in basis, keep track of DCA performance

### ISSUE THREAD

**Issue by Jorijn** [Jorijn/bitcoin-dca#36]
Someone opted the idea to let Bitcoin DCA report how your DCA strategy is performing. Ideally the app would write the transactions to some sort of database and when the user would request statistics, fetch the current price and show how much % the user has progressed.

**Comment by Sebastix** [comment]
There is already some sort of logging build into the app right? When you provide the tag (`-t`) argument to the buy command, the buy is attached to a tag (https://bitcoin-dca.readthedocs.io/en/latest/tagged-balance.html#tagged-balance). This info is written down in a JSON file in de application dir `/var/storage/balance.db`
My idea:
- Add enviroment variable (boolean) to enable logging all buy transactions (or call it a portfolio feature maybe)
- When enabled, a new record is written into a JSON with the amount of bought btc and the timestamp (and maybe also the price in fiat you paid)
- To check the performance overall, an interface (application) would be ideal which reads all the info from the json and a API is used to get the current btc price. This interface/application could be fully client-side, so you can easily open this at anytime without any use of Docker.

**Comment by Jorijn** [comment]
I like this idea. Yet, isn't SQLite better? Using JSON as a (growing) database isn't very efficient since the code has to parse the entire file every time a line should be added. This would also help in the event of a web interface as SQLite offers basic query functions like `GROUP BY` and `SUM`, offloading this logic to presumably more efficient (than PHP) native functions.

**Comment by Sebastix** [comment]
I agree using a SQLite database is better, but then we need a Docker container to run this database I think?
Another question arises to me, is it possible to query this database from the client / application (Javascript)? Make use of an library like https://github.com/sql-js/sql.js (I don't have expierence with that)? Or perhaps using [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) is an option?
